### DISCUSSION TITLE
Wall-clock and Off-CPU profiling go-test functions. Plus Thanks!

### DISCUSSION THREAD

**Discussion by cameronelliott** [felixge/fgprof#13] [Show and tell]
Hey @felixge, thanks for creating this package 🥳 , it's very hard to find examples of how to 
do basic, old-fashion Off-CPU profiling, without starting from scratch with the pprof package.
This package makes it straightforward to do this type of profiling. 
So I really appreciate your work in creating this.

One suggestion, you might show a simple example of using
the package without HTTP for profiling test methods.

I got confused that your package only works with HTTP profiling, where
I just wanted to Start() profiling, and stop it at the end of a TestXXX(t *t.testing) function.
(Rather than controlling duration using HTTP)

So a simple example (for Readme??) might help others that need to do wall-clock or Off-CPU profiling of go-test functions.

```go
package main

import (
	"os"
	"testing"
	"time"
	"github.com/felixge/fgprof"
)

func foo() {
	time.Sleep(time.Second * 4)
}
func TestFunc(t *testing.T) {

	f, err := os.Create("out.pprof")
	if err != nil {
		panic(err)
	}
	defer fgprof.Start(f, fgprof.FormatPprof)()

	foo()
}
```
To view the profile use (where ./m is the binary):
```bash
go tool pprof -http=:8080 ./m out.pprof
```

**Comment by felixge** [comment]
Good idea. I'll try to add this to the README.
